Remove automatic migration execution; support via configuration instead

stevejalim opened the pull request:
This changeset removes all code which previously autodetected the need to run migrations and then call `manage.py migrate` on the target app.  This behaviour did not play well with pipelined deployments that features multiple apps in a node.

Instead, post-deployment tasks (migrations being the key example) must now be specified via configuration, as a list of shell commands to run via a `post_deploy` node in the config YAML file.

See `settings/sample.conf` for an example of this.

It has been tested locally, but no code test coverage has been added. It can be, if necessary, but the changes are relatively clear

Other changes this squashed commit

* version bump to 0.3 - this is a breaking change
* add a changelog file
* update gitignore to exclude build, dist and coverage dirs

This changeset addresses #10 (and, indirectly #3 and #4). All three can be closed when this is merged

Commit:
Remove automatic migration detection and execution; support migrations via configuration instead

This changeset removes all code which previously autodetected the need to run migrations and then call `manage.py migrate` on the target app.  This behaviour did not play well with pipelined deployments that features multiple apps in a node.

Instead, post-deployment tasks (migrations being the key example) must now be specified via configuration, as a list of shell commands to run via a `post_deploy` node in the config YAML file.

See `settings/sample.conf` for an example of this.

Other changes this squashed commit

* version bump to 0.3 - this is a breaking change
* add a changelog file
* update gitignore to exclude build, dist and coverage dirs

hugorodgerbrown commented on heroku_tools/settings/sample.conf:
```diff
@@ -31,6 +31,16 @@ application:
+    # Heroku app with --app or -a as if on the CLI.
+
+    post_deploy:
+        - heroku run python manage.py migrate -a live_app --noinput
```
Is the `heroku run` here required - would there be any non-heroku commands run?

stevejalim replied:
Probably not, but we would need the app to be explicitly referenced with -a/-app rather than using the `application` from the settings, because that references just one app, not all apps in the pipeline. Given that, it seemed more flexible to expect user to enter the full CLI args

hugorodgerbrown replied:
Agreed. Have cleaned out some old cruft and pushed an update.

stevejalim commented:
A follow-up piece (or could be rolled in here) is to allow the user to say Y/N to every item in the post-deploy tasks, so that one wouldn't _have_ to run them every single time

Commit:
Remove collectstatic and redundant configuration settings

add_rich_tag and release_note aren't used, and collectstatic is no longer
relevant now that we have post_deploy commands.

hugorodgerbrown commented:
Released as 0.3.0
